### DISCUSSION TITLE
Vitest in browser fails with Cannot read properties of undefined (reading 'toStringTag')

### DISCUSSION THREAD

**Discussion by SebastianStehle** [vitest-dev/vitest#4342] [Q&A]
Hi,

 I am a porting a project from custom webpack to vite and vitest. The test run fine without browser mode (there are some issues I have to fix), but I need the DOM for some tests. Therefore I would like to test with browser mode enabled.

When I am running the tests I get the following error for almost any test:

```
src/core/utils/color-palette.spec.ts
TypeError: Cannot read properties of undefined (reading 'toStringTag')
 - /node_modules/.vite/deps/@svgdotjs_svg__js.js:1:33
 ```

But when I check the file that is mentioned, there is no call of toStringTag. But I realized that when I test it in chrome, there is this call above every file:

```
const __vi_inject__ = {
    [Symbol.toStringTag]: "Module"
};
```

So it must have been added on the fly by the dev server:

![image](https://github.com/vitest-dev/vitest/assets/1236435/b03d011e-e3c1-47a2-85be-3ee112757550)

Tbh I don't know what this is and why it has been added there and so far I could not find any solution. Not sure if it is a bug or something else, e.g. some problem in my dependencies. I think it has probably nothing to do with svg.js, it is just the first file.

**Comment by sheremet-va** [comment]
Try updating the browser. Looks like yours is too old.

You can also disable this behavior with [`browser.slowHijackESM: false`](https://vitest.dev/config/#browser-slowhijackesm)

**Reply by SebastianStehle** [reply]
Hi,

thanks for your fast answer. My Chrome browser was up to date. But the setting helps :)
